cs/src/core/Allocator/GenericAllocator.cs

```diff
@@ -191,6 +191,13 @@ internal override bool TryComplete()
         /// </summary>
         public override void Dispose()
         {
+            // Dispose base allocator
+            base.Dispose();
```

Comment:
It's generally better to destruct the base after derived. Did this need to move here?